L1Trigger/TrackFindingTracklet/src/TrackletConfigBuilder.cc

```diff
@@ -1364,6 +1511,73 @@ void TrackletConfigBuilder::writeVMSMemories(std::ostream& os, std::ostream& mem
   }
 }
 
+void TrackletConfigBuilder::writeVMSMemoriesExt(std::ostream& os, std::ostream& memories, std::ostream&) {
+  string layerdisk1_, layerdisk2_, layerdisk3_;
+  map<string, int> mem_tracker;
+
+  for (auto& [seed, seed_info] : seedwires_.items()) {
+    // set layer/disk types based on input seed name
+    setLayDiskStr(layerdisk1_, layerdisk2_, layerdisk3_, seed);
+    for (auto& [region, region_info] : seed_info.items()) {  //TPD instance
+      for (string mem : region_info["innerstub"]) {          // write inner/third stub memories
+
+        // add to the memory tracker
+        string mem_idx = layerdisk3_ + "PHI" + mem;
+        if (mem_tracker.find(mem_idx) == mem_tracker.end()) {
+          mem_tracker[mem_idx] = 1;
+        } else {
+          mem_tracker[mem_idx]++;
+        }
+
+        // for overlap region seeds
+        string mem_large = mem.substr(0, 1);
+        if (mem_large == "x")
+          mem_large = "A";
+        else if (mem_large == "y")
+          mem_large = "B";
+        else if (mem_large == "z")
+          mem_large = "C";
+        else if (mem_large == "w")
+          mem_large = "D";
+
+        os << "VMSTE_" << layerdisk3_ << "PHI" << mem << "n" << mem_tracker[mem_idx] << " input=> VMR_" << layerdisk3_
+           << "PHI" << mem_large << ".vmstubout_seed_" << strSeedInt(seed) << " output=> TPD_" << seed << region
+           << ".thirdvmstubin" << std::endl;
+        memories << "VMStubsTE: VMSTE_" << layerdisk3_ << "PHI" << mem << "n" << mem_tracker[mem_idx] << " [18]"
+                 << std::endl;
+      }
+
+      for (string mem : region_info["outerstub"]) {  // write outer/second stub memories
+
+        // add to the memory tracker
+        string mem_idx = layerdisk2_ + "PHI" + mem;
+        if (mem_tracker.find(mem_idx) == mem_tracker.end()) {
+          mem_tracker[mem_idx] = 1;
+        } else {
+          mem_tracker[mem_idx]++;
+        }
+
+        // for overlap region seeds
+        string mem_large = mem.substr(0, 1);
```

Comment:
This looks rather long winded. Why not convert the 1 character long string to upper case with something like:
string s_up{char(std::toupper(s_down[0]))};
where the outer set of brackets are curly ones.

Reply:
Done. (I'm following up, since @cgsavard is gone.)